docs/docs/kb/12-plugins.md

```diff
@@ -0,0 +1,140 @@
+---
+sidebar_position: 12
+description: Scaffolding and interacting with plugins.
+---
+
+# Plugins
+
+Ignite plugins offer a way to extend the functionality of the Ignite CLI. There are two core concepts within plugins `Plugin Commands` and `Hooks`. Where commands extend the cli's functionality, and `hooks` extend existing command functionality. Plguins can be `registered` in an Ignite scaffolded Blockchain project through the `config.yml`
+
+# Adding plugins to a project
+Plugins are registered per project. When a command is executed within a directory on disk which contains an configuration file with a plugin registered. Ignite will automatically load it into itself before a command is run. If the plugin has not already been built there will be a build step before the command is executed whcih the cli will indicate. Below is an example configuration registering a plugin:
+
+``` yaml
+plugins:
+    # must be the name same name as the plugin project name
+  - plugin:
+    # path the plugin project directory 
+    # path can be a remote git repository or local directory
+    path: /path/to/plugin
+    # parameters which can be passed to commands when executing
+    with:
+        foo: value
+        bar: value
+```
+
+## Plugin Commands
+Plugin commands are custom commands added to the ignite cli by a `registered` plugin. Commands can be of any path not defined already by ignite. All plugin commands will extend of the command root `ignite`. An example command definition is outlined below:
+```go
+func (p) Commands() []plugin.Command {
+	// TODO: write your command list here
+	return []plugin.Command{
+        {
+            Use:               "plugin",
```

Comment:
Bad example, because `ignite plugin` is already an existing ignite command :)

Reply:
good catch, will change